scripts/operatorhub-deploy.sh

```diff
@@ -0,0 +1,138 @@
+# https://twitter.com/GitHubHelp/status/322818593748303873
+(
+    cd "${eunomia_root}/deploy/olm-catalog" &&
+        svn checkout https://github.com/operator-framework/community-operators/trunk/upstream-community-operators/eunomia &&
+        rm -rf "${eunomia_root}/deploy/olm-catalog/eunomia/.svn" # Delete artifacts
+)
+echo -e "${WHITE}Files successfully generated${NC}"
+
+# Find latest version
+latest=$(grep currentCSV "${eunomia_root}/deploy/olm-catalog/eunomia/eunomia.package.yaml" | awk -F"eunomia.v" '{print $2}')
+
+# Generate CSV of a new version
+echo -e "\n${WHITE}Generating a new ClusterServiceVersion yaml file${NC}"
+(
+    cd "$eunomia_root" &&
+        operator-sdk olm-catalog gen-csv --csv-version "$new_version" --from-version "$latest"
+)
+# By default copy gitopsconfig crd from latest version
+cp "${eunomia_root}/deploy/olm-catalog/eunomia/${latest}/gitopsconfigs.eunomia.kohls.io.crd.yaml" \
+    "${eunomia_root}/deploy/olm-catalog/eunomia/${new_version}/"
+
+# Make currentCSV in eunomia.package.yaml point to the new version
+# Why make bkup and remove it just after:
+# https://stackoverflow.com/questions/5694228/sed-in-place-flag-that-works-both-on-mac-bsd-and-linux
+sed -i.bkup s/eunomia\.v.*$/eunomia\.v${new_version}/g "${eunomia_root}/deploy/olm-catalog/eunomia/eunomia.package.yaml" &&
+    rm -f "${eunomia_root}/deploy/olm-catalog/eunomia/eunomia.package.yaml.bkup"
+echo -e "${WHITE}ClusterServiceVersion yaml file successfully generated${NC}"
+
+# Stop execution of the script for the user to be able to verify, or modify the files
+cat <<EOT
+
+Check if everything is alright with the new Eunomia OperatorHub release.
+If you want to change something in the description, edit the new CSV file
+that can be found in deploy/olm-catalog/eunomia/${new_version} directory.
+
+To see how it will look like in OperatorHub, paste the contents of
+the newly-generated CSV file to https://operatorhub.io/preview
+
+EOT
+read -p "Press Enter to continue"
+
+# Verify the generated CSV
+echo -e "\n${WHITE}Verifying the newly-generated CSV${NC}"
+operator-courier --verbose verify "${eunomia_root}/deploy/olm-catalog/eunomia"
+operator-courier --verbose verify --ui_validate_io "${eunomia_root}/deploy/olm-catalog/eunomia"
+echo -e "${WHITE}Verification successful${NC}"
+
+echo -e "\n${WHITE}Creating a Pull Request in OperatorHub${NC}"
+branch_name="eunomia-v${new_version}"
+git_context="git -C ${TMPDIR}/community-operators"
+hub_context="hub -C ${TMPDIR}/community-operators"
+
+git -C "$TMPDIR" clone https://github.com/operator-framework/community-operators
+$hub_context fork --remote-name tmp
+$git_context checkout -b "$branch_name"
+cp -R -f "${eunomia_root}/deploy/olm-catalog/eunomia" "${TMPDIR}/community-operators/upstream-community-operators/"
+$git_context add .
+$git_context commit -m "Eunomia release $new_version" --signoff
+$git_context push --force tmp
+$hub_context pull-request -m "Eunomia release $new_version"
```

Comment:
[optional] Could we add a `--dry-run` option, which would avoid running this step? or, alternatively, only do the PR if user explicitly provides a `--send-pr` option? Or, even more alternatively, don't do this line at all, and just print a command that user should call to finalize sending the PR?

Reply:
I agree with @akavel using the `hub` command to create the pull request need to be optional.